### PR TITLE
Unify log messages

### PR DESCRIPTION
**Pull request by danewalton**

### CODE REVIEW

File: source/azure_iot_hub_client.c [hunk 1]
```diff
@@ -665,7 +665,7 @@ AzureIoTHubClientResult_t AzureIoTHubClient_SetSymmetricKey( AzureIoTHubClient_t
         ( pucSymmetricKey == NULL ) || ( ulSymmetricKeyLength == 0 ) ||
         ( xHMACFunction == NULL ) )
     {
-        AZLogError( ( "AzureIoTHubClient_SetSymmetricKey failed: Invalid argument" ) );
+        AZLogError( ( "AzureIoTHubClient_SetSymmetricKey failed: invalid argument" ) );
```
Comment (hihigupt) [inline source/azure_iot_hub_client.c:668]: >" [](start = 81, length = 1)

I see full stop at the end of comment and some places not. can we be consistent with that 

Line:698

**Comment by hihigupt** [comment]
>         AZLogDebug( ( "Attempt to unsubscribe from the MQTT topic: %s.", AZ_IOT_HUB_CLIENT_METHODS_SUBSCRIBE_TOPIC ) );

Attempting 

---
Refers to: source/azure_iot_hub_client.c:1041 in deab823. [](commit_id = deab823ba37cc7592948d28e3a3f181286beed04, deletion_comment = False)

**Comment by hihigupt** [comment]
>     AZLogDebug( ( "AzureIoTProvisioning updated state from [%d] -> [%d]", ulState,

%u?

---
Refers to: source/azure_iot_provisioning_client.c:132 in deab823. [](commit_id = deab823ba37cc7592948d28e3a3f181286beed04, deletion_comment = False)

**Comment by hihigupt** [comment]
>     AZLogDebug( ( "AzureIoTProvisioning updated state from [%d] -> [%d]", ulState,

all the places we print state

---
In reply to: [820742446](https://github.com/Azure/azure-iot-middleware-freertos/pull/90#issuecomment-820742446) [](ancestors = 820742446)

---
Refers to: source/azure_iot_provisioning_client.c:132 in deab823. [](commit_id = deab823ba37cc7592948d28e3a3f181286beed04, deletion_comment = False)

**Comment by hihigupt** [comment]
>     AZLogDebug( ( "AzureIoTProvisioning updated state from [%d] -> [%d]", ulState,

Suggestion: We can have function that converts state number to const string, and only use that in logging. So if we turn of logging we remove those const strings.

---
In reply to: [820742829](https://github.com/Azure/azure-iot-middleware-freertos/pull/90#issuecomment-820742829) [](ancestors = 820742829,820742446)

---
Refers to: source/azure_iot_provisioning_client.c:132 in deab823. [](commit_id = deab823ba37cc7592948d28e3a3f181286beed04, deletion_comment = False)

File: source/azure_iot_provisioning_client.c [hunk 2]
```diff
@@ -243,7 +243,7 @@ static void prvProvClientSubscribe( AzureIoTProvisioningClient_t * pxAzureProvCl
         if( ( xMQTTResult = AzureIoTMQTT_Subscribe( &( pxAzureProvClient->_internal.xMQTTContext ),
                                                     &xMQTTSubscription, 1, usSubscribePacketIdentifier ) ) != eAzureIoTMQTTSuccess )
         {
-            AZLogError( ( "AzureIoTProvisioning failed to SUBSCRIBE to MQTT topic. Error=%d", xMQTTResult ) );
+            AZLogError( ( "AzureIoTProvisioning failed to SUBSCRIBE to MQTT topic: MQTT error=0x%08x", xMQTTResult ) );
```
Comment (hihigupt) [inline source/azure_iot_provisioning_client.c:246]: >SUBSCRIBE [](start = 58, length = 9)

SUBSCRIBE -> subscribe

**Comment by hihigupt** [comment]
>         xResult = eAzureIoTProvisioningFailed;

should we log core error?

---
Refers to: source/azure_iot_provisioning_client.c:284 in deab823. [](commit_id = deab823ba37cc7592948d28e3a3f181286beed04, deletion_comment = False)

File: source/azure_iot_provisioning_client.c [hunk 3]
```diff
@@ -413,7 +413,7 @@ static void prvProvClientParseResponse( AzureIoTProvisioningClient_t * pxAzurePr
         }
         else if( pxAzureProvClient->_internal.xRegisterResponse.retry_after_seconds == 0 )
         {
-            AZLogError( ( "AzureIoTProvisioning client failed with error %d, extended error status: %u and no server retry time duration",
+            AZLogError( ( "AzureIoTProvisioning client registration failed with error %d, extended error status: %u and no server retry time duration",
```
Comment (hihigupt) [inline source/azure_iot_provisioning_client.c:416]: >d [](start = 87, length = 1)

%u

**Comment by hihigupt** [comment]
>             AZLogDebug( ( "AzureIoTProvisioning is in complete state, with status :%d",

0x%08x

---
Refers to: source/azure_iot_provisioning_client.c:528 in deab823. [](commit_id = deab823ba37cc7592948d28e3a3f181286beed04, deletion_comment = False)

**Comment by hihigupt** [comment]
>             AZLogError( ( "AzureIoTProvisioning  failed to process loop: ProcessLoopDuration=%u, Error=%d",

0x%08x

---
Refers to: source/azure_iot_provisioning_client.c:536 in deab823. [](commit_id = deab823ba37cc7592948d28e3a3f181286beed04, deletion_comment = False)

File: source/azure_iot_provisioning_client.c [hunk 4]
```diff
@@ -931,7 +931,7 @@ AzureIoTProvisioningClientResult_t AzureIoTProvisioningClient_GetDeviceAndHub( A
 
         if( ( *pulHostnameLength < ulHostnameLength ) || ( *pulDeviceIDLength < ulDeviceIDLength ) )
         {
-            AZLogWarn( ( "AzureIoTProvisioning Memory buffer passed is not enough to store hub info" ) );
+            AZLogWarn( ( "AzureIoTProvisioning memory buffer passed is not enough to store hub info" ) );
```
Comment (hihigupt) [inline source/azure_iot_provisioning_client.c:934]: this is Error

**Review by hihigupt — APPROVED** [review]

File: source/azure_iot_hub_client.c (hunk 1 above)
Comment (danewalton) [reply source/azure_iot_hub_client.c:668]: Freertos looks like it doesn't so I'll remove them

**Comment by danewalton** [comment]
> > ```
> > AZLogDebug( ( "AzureIoTProvisioning updated state from [%d] -> [%d]", ulState,
> > ```
> 
> all the places we print state
> 
> In reply to: [820742446](https://github.com/Azure/azure-iot-middleware-freertos/pull/90#issuecomment-820742446) [](ancestors = 820742446)
> 
> Refers to: source/azure_iot_provisioning_client.c:132 in [deab823](https://github.com/Azure/azure-iot-middleware-freertos/commit/deab823ba37cc7592948d28e3a3f181286beed04). [](commit_id = [deab823](https://github.com/Azure/azure-iot-middleware-freertos/commit/deab823ba37cc7592948d28e3a3f181286beed04), deletion_comment = False)

updated all of those to be `[%u]`

**Comment by danewalton** [comment]
> > ```
> > AZLogDebug( ( "AzureIoTProvisioning updated state from [%d] -> [%d]", ulState,
> > ```
> 
> Suggestion: We can have function that converts state number to const string, and only use that in logging. So if we turn of logging we remove those const strings.
> 
> In reply to: [820742829](https://github.com/Azure/azure-iot-middleware-freertos/pull/90#issuecomment-820742829) [](ancestors = 820742829,820742446)
> 
> Refers to: source/azure_iot_provisioning_client.c:132 in [deab823](https://github.com/Azure/azure-iot-middleware-freertos/commit/deab823ba37cc7592948d28e3a3f181286beed04). [](commit_id = [deab823](https://github.com/Azure/azure-iot-middleware-freertos/commit/deab823ba37cc7592948d28e3a3f181286beed04), deletion_comment = False)

Yea I think this can be a stretch goal with some more thinking about how to organize that.

**Pull request merged**